plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -392,13 +406,16 @@ def deploy(release_docs)
       Samson::Parallelizer.map(resources, db: true, &:deploy)
     end
 
-    def deploy_and_watch(release_docs, timeout)
+    def deploy_and_watch(release_docs, timeout, show_logs_on_success)
```

Comment:
show_logs_on_success is not a logical 3rd argument, so should use keyword args

`, show_logs_on_success:` and then call with `show_logs_on_success: true/false`

Reply:
also the flag does not make it show logs, just 'maybe' show logs, so maybe better to use `prerequisite: true`

Reply:
... we can also show logs for pods that are not prerequisites if the user asked for them ... so don't need this flag at all

Reply:
Sure! I'll change it to keyword args, thanks for the tips (really valuable for non rubyst :))

Do you think we don't need a third flag? Or a flag to only show prerequisites and show them all (i.e. without the pod.completed? filter)?

I changed it to the latter. The reason is that if we print the same after the deploy (not prerequisites, of deploys deploys) ends, it will still find the logs of the prerequisites pods and it will be printed again. I'm not sure how to filter those, nor if it's worth it.

But if we do the print after the deploys ends, without any more filters, then if something was print after the prerequisites run, then it will be printed again. And that seems confusing, it's just a duplicated print (most of the times).

Maybe after the deploys deploys, we can print but excluding pods that were completed when we printed the first time. But it seems a little messier than I'd like, and not sure the functionality it's worth the complexity.

Also, after the deploys ends, most of the time you deploy some always living objects (like deployments). The ones that are short-lived are probably printed in the first interruption, after prerequisites deploys, so there is no much gain, I think, printing after the deploy ends again.

What do you think?